File: 03 Berechnungen/Go/Aufgabe4.go
package main

import "fmt"

var (
	h                  int
	m                  int
	s                  int
	sekSeitMitternacht int
	anteil             float32
)

func main() {
	fmt.Print("Gib die Stunden ein: ")
	fmt.Scan(&h)

	fmt.Print("\nGib die Minuten ein: ")
	fmt.Scan(&m)

	fmt.Print("\nGib die Sekunden ein: ")
	fmt.Scan(&s)
	fmt.Println()

	fmt.Printf("Eingegebene Zeit: %d:%d:%d\n", h, m, s)

	sekSeitMitternacht = s + 60*m + 60*60*h

	fmt.Printf("Anzahl der Sekunden seit Mitternacht: %d\n", sekSeitMitternacht)

	fmt.Printf("Anzahl der noch verbleibenden Sekunden: %d\n", 60*60*24-sekSeitMitternacht)

	anteil = float32(sekSeitMitternacht) / float32(60*60*24)

	fmt.Printf("Es sind schon %.2f%% des Tages verstrichen.", anteil*100)
}

File: 03 Berechnungen/Pascal/Aufgabe4.pas
Program Aufgabe4;

{$codepage utf8}

Var H, M, S, SekSeitMitternacht: LongInt;
    Anteil: Real;

Begin
    Write('Gib die Stunden ein: ');
    ReadLn(H);

    Write('Gib die Minuten ein: ');
    ReadLn(M);

    Write('Gib die Sekunden ein: ');
    ReadLn(S);

    WriteLn('Eingegebene Zeit: ', H, ':', M, ':', S);

    SekSeitMitternacht := S + M * 60 + H * 60 * 60;

    WriteLn('Anzahl der Sekunden seit Mitternacht: ', SekSeitMitternacht);

    WriteLn('Anzahl der noch verbleibenden Sekunden des Tages: ', 60 * 60 * 24 - SekSeitMitternacht);

    Anteil := SekSeitMitternacht / (60 * 60 * 24);

    WriteLn('Es sind schon ', (Anteil * 100):2:2, '% des Tages verstrichen.');
End.